datalad/__init__.py

```diff
@@ -12,6 +12,14 @@
 
 # Other imports are interspersed with lgr.debug to ease troubleshooting startup
 # delays etc.
+
+# If there is a bundled git, make sure GitPython uses it too:
+from datalad.cmd import GitRunner
+_git_runner = GitRunner()
+if _git_runner._GIT_PATH:
+    import os
+    os.environ['GIT_PYTHON_GIT_EXECUTABLE'] = os.path.join(_git_runner._GIT_PATH, 'git')
```

Comment:
Because it isn't relevant just in gitrepo.py. ;-)
First of all, I think there's still some code, especially in the tests, that uses GitPython directly without instantiating `GitRepo`.
But more importantly, if we access `GitRepo`'s underlying GitPython objects, git is called by GitPython, not by our `Runner`. So there's nothing the `GitRepo` instance on top of it could do about it, except setting this environment variable in `GitRepo`s constructor. And this is pointless, since it nevertheless would be a universal setting, which has nothing to do with a certain repository. Plus: We would do it over and over again for no reason. Plus: We loose consistency, when using GitPython directly. So I don't see no advantage, but only disadvantages in moving this to `GitRepo`.